src/Microsoft.AspNetCore.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -14,6 +15,9 @@ namespace Microsoft.AspNetCore.Razor.Parser.TagHelpers.Internal
 {
     public class TagHelperParseTreeRewriter : ISyntaxTreeRewriter
     {
+        // Newlines are invalid markup for HTML attribute values.
+        private static readonly string InvalidAttributeValueMarker = Environment.NewLine;
```

Comment:
Hmm. Alright, looks like the null unicode character is invalid. Swapped to use that.